### DISCUSSION TITLE
Feature Request Dimensions

### DISCUSSION THREAD

**Discussion by ghost** [input-leap/input-leap#1776] [Feature Requests]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/864
> 
> Issue created on: 2020-09-03 by @nowell-morris
> Issue last updated on: 2021-07-15

I am wondering if we could make our "configure server" aware of a double screen.  Right now I have to go to the right, then up, then left to reach the monitor in the top left.  But since my server is dual screen, could the configuration be aware of that and allow me to go up into the other screens depending upon which screen I am moving from?
Server setup:
![image](https://user-images.githubusercontent.com/53183382/92139315-6d384a80-edc4-11ea-8391-9cc2d650dadd.png)
my screens:
![image](https://user-images.githubusercontent.com/53183382/92139616-d455ff00-edc4-11ea-8725-787188645cdb.png)

---

> Commented on: 2020-09-25 by @Absentity

I'd like this too eventually. As a workaround until something like this exists, you can set this up manually.

(As of Barrier 2.3.3)
1. From the Barrier window menu, go to Barrier > Save Configuration
2. Edit the links section like below. Note that the numbers are percent, 0-100. 0 starts at the top and the left. You might need to tinker.
3. Back in the Barrier window, switch to "Use existing configuration" and put in your modified config.
4. Click Reload
5. Cheers!

This is kind of like what my setup is:
```
# +--------+ +-----------+
# |   A    | |           |
# |        | |           |
# +--------+ |           |
# +--------+ |           |
# |   B    | |   DESK    |
# |        | |           |
# +--------+ |           |
#            +-----------+
#
#
section: links
	B:
		right = DESK(43,55)
		up = A
	A:
		right = DESK(0,43)
		down = B
	DESK:
		left(0,43) = A
		left(43,100) = B	
end
```

Sadly, I don't remember where I learned this. I add "synergy" to my searches sometimes to find what I need.

---

> Commented on: 2020-09-28 by @nowell-morris

Absentity, this is fantastic!  I was able to use this approach and improve my setup.  I can directly shift from client back to server whereas I could not before.  For whatever reason, I still have to connect to clients with my original post details (in a hooked path).  It could be that I need to restart all the machines together, which I have not yet had time to do all at once.  But I will try it out and I will take your suggestion at searching more troubleshooting steps with "synergy" as potential help in finding them.

---

> Commented on: 2020-10-09 by @AcousticRand

A similar issue that cannot be solved by the solution using partials (i.e. parenthesis with % markers):
```
+----++----++----+
| 3A || 1A || 4A |
+----++----++----+
+----++----++----+
| B  || 2A || C  |
+----++----++----+
A=linux Server Ubuntu 20.04
B=linux Client Ubuntu 18.04
C=windows Client
```

In order to get to B or C from computer A, I must move from 2A to 1A, and then left or right across 3A or 4A to get to B or C respectively.  I'm guessing this is because it recognizes my left and right-most edges as 3A and 4A.
So all that to say that I'd appreciate the ability to customize the layouts with monitor granularity (unless I'm missing something obvious, which is entirely possible).

---

> Commented on: 2020-10-09 by @nowell-morris

just as following up - if you look at my original picture of my four monitors, I can do a counter-clockwise circle now through all 4 monitors, but I cannot do the same path in a clockwise fashion.  I can leave the top left monitor down to my server (displays at the bottom) in relation to the % that I designated like Absentity suggested.  But if I try to move upward from my bottom left monitor to the one above it, it will not work.  I did reboot everything.  I do have updated barrier packages on everything.  I made sure that DNS was resolvable by all of the machines in question (being able to look each other up by name and nickname).

---

> Commented on: 2021-07-15 by @NicolasGoeddel

I have a similar issue. I've got this setup:
```
      +-----+
      |  A2 |
+-----+-----+-----+-----+
|  A1 |  A3 |  B1 |  B2 |
+-----+-----+-----+-----+
```
A and B are Ubuntu machines. Every monitor has a resolution of 1920x1080.
If the cursor is on A2 and I go right I end up in the upper half of B1.
If the cursor is on A3 and I go right I end up in the lower half of B2.
And vice versa. This is a bit annoying.

This is what I want:
Going right from A2 should not be possible.
Moving between A3 and B1 should be flawless. So don't map the height of A3 to the lower half of B1.

My guess is that Synergy sees the configuration like this:
```
      +-----+-----------+-----------+
      |  A  |           |           |
+-----+-----+     B1    |     B2    |
|  A  |  A  |           |           |
+-----+-----+-----------+-----------+
```

This is especially annoying if the configuration is like this:
```
      +-----+
      |  A2 |
+-----+-----+-----+-----+
|  A1 |  A3 |  A4 |  B1 |
+-----+-----+-----+-----+
```
Here it behaves like in the first example:
The right edge of A2 is mapped to the upper half of B1.
The right edge of A4 is mapped to the lower half of B1.

This can be really frustrating when you move your cursor from B1 to the left and you land in A2.

---

> Commented on: 2021-07-15 by @NicolasGoeddel

I was able to solve my problem by using the configuration file. But it would still be nice to be able to configure that in the GUI.

---

> Commented on: 2021-07-15 by @NicolasGoeddel

Here is the link to the syntax of the configuration file: https://symless.com/help-articles/creating-text-config-files

**Comment by barrelltitor** [comment]
```
+----++----+
| 1A || 2A |
+----++----+
+----++----+
| 3A  || B |
+----++----+
```
Is there any way to get this configuration working to go down from 2A and right from 3A?
A - Windows with 3 monitors(2 regular ones and a tiny portable one under them)
B - Linux  with 1 monitor 

EDIT:

This seems to semi-work:
```
section: links
	Windows:
		down = barrelltitor-laptop
	barrelltitor-laptop:
		up = Windows(60,90)
end
```
barrelltitor-laptop is the linux machine
If I go up from the laptop, it goes to 2A. However, if I add down(60,90) as such:
```
section: links
	Windows:
		down(60,90) = barrelltitor-laptop
	barrelltitor-laptop:
		up = Windows(60,90)
end
```
I cannot get to the laptop screen in any way. Any ideas?

**Reply by shymega** [reply]
This is not a feature request, please open a support request in 'Discussions' if it's a configuration issue, or if a actual bug, then a bug report.
